### PR TITLE
Moves the random admin folder name generation at the end of the install process

### PR DESCRIPTION
**Pull request by matthieu-rolland**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Due to a modification in the way the path for the admin folder is set, we have a cache issue after an install in production mode, this PR moves the random admin folder name generation at the end of the install process instead of when displaying the BO login page for the first time. Both in UI and CLI install.
| Type?             | refacto
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | 1/ Generate a release from this PR <br> 2/ Install the release manually <br> 3/ It should work as before. <br> 4/ Same test with CLI install, from a release from this PR as well 
| UI Tests          | [click here](https://github.com/matthieu-rolland/ga.tests.ui.pr/actions/runs/8850250859)| Fixed issue or discussion?     | 
| Related PRs       | 
| Sponsor company   | PrestaShop SA

### Review guide:

Before this PR, in production, the admin folder name is not generated at install, it is generated when we first display the BO login page. 
With this PR, the admin folder name is generated at the end of the install process

In a nutshell, this PR dos:

- Add an install process called "Finalization", that takes care of creating the admin folder *
- Move the admin folder detection from AppKernel.php in a dedicated class, so that it can be used at install process

*The random admin folder name is generated at the beginning of the install process so that the name of the folder is available in the final install page, where the link is displayed, because this page is generated before the install process.

### CODE REVIEW

**Comment by matthieu-rolland** [comment]
I've seen the phpstan issue, I'll fix it

**Review by tleon — DISMISSED** [review]

**Review by boherm — DISMISSED** [review]

**Review by M0rgan01 — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Review by jolelievre — DISMISSED** [review]
Thanks @matthieu-rolland

**Review by matks — DISMISSED** [review]

**Review by jolelievre — DISMISSED** [review]
Thanks @matthieu-rolland

**Comment by matthieu-rolland** [comment]
Latest UI tests: https://github.com/matthieu-rolland/ga.tests.ui.pr/actions/runs/8830417094

**Review by AureRita — DISMISSED** [review]
Hi @matthieu-rolland 

Thank you for your PR, I tested it with you and we seen that works as expected  

Link to the auto test : https://github.com/matthieu-rolland/ga.tests.ui.pr/actions/runs/8830417094

Because the auto test is :green_circle: and the PR seems to works as expected, It's QA :heavy_check_mark: 

Thank you

**Comment by jolelievre** [comment]
I made a little modification because the output in CLI didn't include the physical URI, all good now:
Capture d’écran 2024-04-26 à 16 53 32

**Review by jolelievre — APPROVED** [review]

**Comment by matthieu-rolland** [comment]
thank you @jolelievre !  here's the latest ui test: https://github.com/matthieu-rolland/ga.tests.ui.pr/actions/runs/8850250859

**Review by AureRita — APPROVED** [review]
Hi @matks 

Thank you for your PR, I tested it and it seems to works, except the Print of the new name of the admin as you can see : 

image
 
If this event is good for you, It'll be QA :heavy_check_mark: 

Link to the auto test : https://github.com/matthieu-rolland/ga.tests.ui.pr/actions/runs/8850250859

Because the auto test is :green_circle: and the PR seems to works as expected, It's QA :heavy_check_mark: 

Thank you
